--- a/tensorflow/lite/core/interpreter.cc
+++ b/tensorflow/lite/core/interpreter.cc
@@ -445,8 +445,10 @@
 }
 
 TfLiteStatus Interpreter::ReportTelemetrySettings(const char* setting_name) {
+#ifdef TF_LITE_TENSORFLOW_PROFILER
   telemetry::TelemetryReportSettings(context_, setting_name,
                                      telemetry_data_.get());
+#endif //TFLITE_TENSORFLOW_PROFILER
   return kTfLiteOk;
 }
 
--- a/tensorflow/lite/core/subgraph.cc
+++ b/tensorflow/lite/core/subgraph.cc
@@ -1382,7 +1382,9 @@
 
 TfLiteStatus Subgraph::Invoke() {
   auto status = InvokeImpl();
+#ifdef TF_LITE_TENSORFLOW_PROFILER
   telemetry::TelemetryReportEvent(&context_, "Invoke", status);
+#endif //TFLITE_TENSORFLOW_PROFILER
   return status;
 }
 TfLiteStatus Subgraph::InvokeImpl() {
@@ -1966,7 +1968,9 @@
 
 TfLiteStatus Subgraph::ModifyGraphWithDelegate(TfLiteDelegate* delegate) {
   auto status = ModifyGraphWithDelegateImpl(delegate);
+#ifdef TF_LITE_TENSORFLOW_PROFILER
   telemetry::TelemetryReportEvent(&context_, "ModifyGraphWithDelegate", status);
+#endif //TFLITE_TENSORFLOW_PROFILER
   return status;
 }
 
